src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,38 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.ValidationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_USERS_AGE = 18;
+    private static final int MIN_PASSWORD_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (validation(user)) {
+            return storageDao.add(user);
+        }
+        throw new ValidationException("This user can`t be registered");
+    }
+
+    private boolean validation(User user) {
+        if (user == null) {
+            throw new ValidationException("User can`t be null");
+        }
+
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new ValidationException("User with this login exist in storage");
+        }
+
+        if (user.getAge() < MIN_USERS_AGE) {
+            throw new ValidationException("User`s age can`t be less than 18");
+        }
+
+        if (user.getPassword().length() < MIN_PASSWORD_LENGTH) {
+            throw new ValidationException("User`s password length can`t be less than 6");
+        }
+        return true;
```

Comment:
also check if age is not null (it's object, not primitive), if password and login are not null/empty